core/silkworm/rlp/encode_vector.hpp

```diff
@@ -0,0 +1,49 @@
+/*
+   Copyright 2020-2021 The Silkworm Authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+#ifndef SILKWORM_RLP_ENCODE_VECTOR_HPP_
+#define SILKWORM_RLP_ENCODE_VECTOR_HPP_
+
+#include <vector>
+
+#include <silkworm/rlp/encode.hpp>
+
+namespace silkworm::rlp {
+
+template <class T>
+size_t length(const std::vector<T>& v) {
+    size_t payload_length{0};
+    for (const T& x : v) {
+        payload_length += length(x);
+    }
+    return length_of_length(payload_length) + payload_length;
+}
+
+template <class T>
+void encode(Bytes& to, const std::vector<T>& v) {
+    Header h{true, 0};
+    for (const T& x : v) {
+        h.payload_length += length(x);
+    }
+    encode_header(to, h);
```

Comment:
to.reserve(to.size() + h.payload_length()); // avoid multiple memory allocations

Reply:
actually line should be before encode_header()

Reply:
Good suggestion! Done in https://github.com/torquem-ch/silkworm/pull/492/commits/c3f994b303b254e913c240ef43929b1ec94e59ff after `encode_header()` because `h.payload_length` doesn't include the length of the RLP header itself.